src/Framework/MockObject/Matcher/ConsecutiveParameters.php

```diff
@@ -58,41 +63,85 @@ public function toString(): string
     }
 
     /**
-     * @throws \PHPUnit\Framework\ExpectationFailedException
+     * @param BaseInvocation $invocation
      *
+     * @throws \PHPUnit\Framework\ExpectationFailedException
      * @return bool
      */
     public function matches(BaseInvocation $invocation)
     {
         $this->invocations[] = $invocation;
         $callIndex           = \count($this->invocations) - 1;
+        $this->parameterVerificationResults[$callIndex] = null;
+
+        if (!$this->shouldInvocationBeVerified($callIndex)) {
+            return false;
+        }
 
-        $this->verifyInvocation($invocation, $callIndex);
+        $this->parameterVerificationResults[$callIndex] =
+            $this->verifyInvocation($invocation, $callIndex);
 
-        return false;
+        return $this->parameterVerificationResults[$callIndex];
     }
 
-    public function verify(): void
+    /**
+     * Verify all constraints within this expectation
+     *
+     * @return bool|mixed
+     */
+    public function verify()
     {
         foreach ($this->invocations as $callIndex => $invocation) {
-            $this->verifyInvocation($invocation, $callIndex);
+
+            if (!$this->shouldInvocationBeVerified($callIndex)) {
+                continue;
+            }
+
+            try {
+                $this->parameterVerificationResults[$callIndex] = $this->verifyInvocation($invocation, $callIndex);
+
+                return $this->parameterVerificationResults[$callIndex];
+            } catch (ExpectationFailedException $e) {
+                $this->parameterVerificationResults[$callIndex] = $e;
+
+                throw $this->parameterVerificationResults[$callIndex];
+            }
         }
     }
 
     /**
-     * Verify a single invocation
+     * Check if the invocation should be verified.
+     * As soon as
+     *  1. this should be verified to avoid duplicate calls when a callback is specified
+     *  2. check is performed for all _and_ individual callbacks
      *
-     * @param int $callIndex
+     * to avoid duplication, it was exported into a separate method
      *
-     * @throws ExpectationFailedException
+     * @param $callIndex
+     * @return bool
```

Comment:
Please do not duplicate the information from type declarations in docblock annotations.